csharp/autobuilder/Semmle.Autobuild.CSharp/DotNetRule.cs

```diff
@@ -100,8 +100,12 @@ public static BuildScript WithDotNet(IAutobuilder<AutobuildOptionsShared> builde
                 }
                 else
                 {
+                    // The .NET SDK was not installed, either because the installation failed or because it was already installed.
                     installDir = null;
-                    env = null;
+                    env = new Dictionary<string, string> {
```

Comment:
This is a small bugfix: if there's a single dotnet SDK on the executing machine, which matches the one specified in `global.json`, then we don't download it again, and we get into this branch of the `if`.